packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -851,17 +830,20 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
     private attachListener() {
         this.setMaxListeners(Number.MAX_SAFE_INTEGER);
         this.dataStoreContext.once("attaching", () => {
-            assert(this.bindState !== BindState.NotBound,
-                0x186 /* "Data store attaching should not occur if it is not bound" */);
-            this._attachState = AttachState.Attaching;
+            assert(
+                this.localAttachState === AttachState.Attached,
+                "Data store should not globally attach if it's not locally attached",
+            );
+            this._globalAttachState = AttachState.Attached;
             // This promise resolution will be moved to attached event once we fix the scheduler.
             this.deferredAttached.resolve();
-            this.emit("attaching");
-        });
-        this.dataStoreContext.once("attached", () => {
-            assert(this.bindState === BindState.Bound,
-                0x187 /* "Data store should only be attached after it is bound" */);
-            this._attachState = AttachState.Attached;
+
+            // Mark all the child contexts as attached.
+            this.localChannelContextQueue.forEach((channel) => {
+                channel.markAttached();
```

Comment:
I think we need to have different verbs for locally & globally attached (unless we merge states into one enum :))
As otherwise reader here need to reason what "attached" case this API name refers to

Reply:
Same with events - "attached" is actually "globally attached", right?